lib/gl_rubocop/gl_cops/unique_identifier.rb

```diff
@@ -0,0 +1,25 @@
+module GLRubocop
+  module GLCops
+    class UniqueIdentifier < RuboCop::Cop::Base
+      MSG = 'View components must include a data-test-id attribute'.freeze
+      EMPTY_MSG = 'data-test-id attribute must not be empty'.freeze
+      UNIQUE_IDENTIFIER = 'data-test-id'.freeze
+
+      # rubocop:disable Metrics/AbcSize
+      def on_send(node)
+        return unless %i[render template].include?(node.method_name)
+        return unless File.exist?(processed_source.file_path)
+
+        raw_content = File.read(processed_source.file_path)
+        return add_offense(node) unless raw_content.include?(UNIQUE_IDENTIFIER)
+
+        line_number = raw_content.lines.find_index { |line| line.include?(UNIQUE_IDENTIFIER) }
+        line = raw_content.lines[line_number]
+
+        test_id_value = line.match(/#{Regexp.quote(UNIQUE_IDENTIFIER)}:\s*["']([^"']*)["']/)[1]
```

Comment:
Similar to the above point, maybe we could store this regex expression in a variable method to reduce complexity within this on_send method